src/EventListener/JWTDecodedListener.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\EventListener;
+
+use App\Repository\UserRepository;
+use Lexik\Bundle\JWTAuthenticationBundle\Event\JWTDecodedEvent;
+
+class JWTDecodedListener
+{
+    /**
+     * @var UserRepository
+     */
+    private $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function onJWTDecoded(JWTDecodedEvent $event)
+    {
+        $payload = $event->getPayload();
+        $user = $this->userRepository->findOneBy(array('username' => $payload['username']));
```

Comment:
As you are searching by username, add a unique constraint in the User entity so you don't have duplicate usernames.

Reply:
This field is already unique in User entity. 
`/**
     * The username of User
     * @ORM\Column(type="string", unique=true)
     * @Serializer\Expose()
     * @Groups({"UserDetail", "UserList"})
     * @SWG\Property()
     */
    private $username;`